rasterio/_env.pyx

```diff
@@ -68,11 +68,11 @@ cdef void log_error(CPLErr err_class, int err_no, const char* msg) with gil:
 # other platforms. Each calls log_error().
 IF UNAME_SYSNAME == "Windows":
     cdef void __stdcall logging_error_handler(CPLErr err_class, int err_no,
-                                              const char* msg):
+                                              const char* msg) with gil:
         log_error(err_class, err_no, msg)
 ELSE:
     cdef void logging_error_handler(CPLErr err_class, int err_no,
-                                    const char* msg):
+                                    const char* msg) with gil:
```

Comment:
Interestingly, this does not fix the behavior for me. How did you identify this as the line that was causing the problem? I use pdb a lot when working with pure python, but haven't got much experience yet with debugging cython. 

Will also try to check out the windows solution before the end of the day.